src/Microsoft.SqlTools.ServiceLayer/LanguageServices/PeekDefinition.cs

```diff
@@ -54,6 +56,8 @@ private Database Database
                     catch(Exception ex)
                     {
                         Logger.Write(LogLevel.Error, "Exception at PeekDefinition Database.get() : " + ex.Message);
+                        this.error = true;
+                        this.errorMessage = SR.PeekDefinitionAzureError;
```

Comment:
This is a dangerous assumption. Many different errors could have caused a general exception. We should only give this kind of message if we're sure.

I'd recommend 2 improvements:
- Always include the actual error message back to the user. The default error should be something like string.Format("An unexpected error occurred during Peek Definition execution: '{0}', ex.Message);
- Catch the most specific exception you can, e.g. ConnectionFailureException. Then _only_ if the target is Azure should you return this message. And you should still return the actual error message too unless you're 100% sure of the cause.